src/taipy/core/config/version_migration_config.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from copy import copy
+from typing import Any, Callable, Dict, Optional
+
+from taipy.config._config import _Config
+from taipy.config.common._template_handler import _TemplateHandler as _tpl
+from taipy.config.config import Config
+from taipy.config.section import Section
+
+from .._version._version_manager import _VersionManager
+
+
+class VersionMigrationConfig(Section):
+    """
+    Holds all the configuration fields needed to register migration functions from an old version to newer one.
+
+    Attributes:
+        id (str): Identifier of the version migration configuration. It must be a valid Python variable name.
+        source_version (str): The version that entities are migrated from.
+        target_version (str): The version that entities are migrated to.
+        data_node_migration_fct (Optional[Callable]): Migration function that takes a DataNode entity from `source_version` as input
+            and returns a compatible DataNode entity with `target_version`. This can be a Python function or None if there is none.
+        task_migration_fct (Optional[Callable]): Migration function that takes a Task from `source_version` as input
+            and returns a compatible Task with `target_version`. This can be a Python function or None if there is none.
+        pipeline_migration_fct (Optional[Callable]): Migration function that takes a Pipeline entity from `source_version` as input
+            and returns a compatible Pipeline entity with `target_version`. This can be a Python function or None if there is none.
+        scenario_migration_fct (Optional[Callable]): Migration function that takes a Scenario entity from `source_version` as input
+            and returns a compatible Scenario entity with `target_version`. This can be a Python function or None if there is none.
+        **properties (dict[str, Any]): A dictionary of additional properties.
+    """
+
+    name = "VERSION_MIGRATION"
+
+    _SOURCE_VERISON_KEY = "source_version"
+    _TARGET_VERISON_KEY = "target_version"
+    _DATA_NODE_MIGRATION_FCT_KEY = "data_node_migration_fct"
+    _TASK_MIGRATION_FCT_KEY = "task_migration_fct"
+    _PIPELINE_MIGRATION_FCT_KEY = "pipeline_migration_fct"
+    _SCENARIO_MIGRATION_FCT_KEY = "scenario_migration_fct"
+
+    def __init__(
+        self,
+        id: str,
+        source_version: str,
```

Comment:
Since the production versions are ordered, I believe the target_version is sufficient. We don't need the source_version. 
Am I right?